pkg/bdb/hash_page.go

```diff
@@ -66,21 +67,22 @@ func HashPageValueContent(db *os.File, pageData []byte, hashPageIndex uint16, pa
 		if err != nil {
 			return nil, xerrors.Errorf("failed to parse page=%d: %w", currentPageNo, err)
 		}
-		if currentPage.PageType != OverflowPageType {
-			continue
+		select {
+		case <-ctx.Done():
+			return nil, xerrors.Errorf("timeout for parse page")
+		default:
+			if currentPage.PageType == OverflowPageType {
+				var hashValueBytes []byte
+				if currentPage.NextPageNo == 0 {
+					// this is the last page, the whole page contains content
+					hashValueBytes = currentPageBuff[PageHeaderSize : PageHeaderSize+currentPage.FreeAreaOffset]
+				} else {
+					hashValueBytes = currentPageBuff[PageHeaderSize:]
+				}
+				hashValue = append(hashValue, hashValueBytes...)
+			}
+			currentPageNo = currentPage.NextPageNo
```

Comment:
No need to indent the whole thing.
you can simply check:
if err := ctx.Err(); err != nil {
     return nil, err
}